src/nupic/bindings/algorithms.i

```diff
@@ -101,9 +101,20 @@ _ALGORITHMS = _algorithms
 #include <nupic/proto/ConnectionsProto.capnp.h>
 #include <nupic/proto/SpatialPoolerProto.capnp.h>
 #include <nupic/proto/TemporalMemoryProto.capnp.h>
+%}
 
+//
+// Numpy API
+//
+%{
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>
+%}
+%init %{
+  import_array();
```

Comment:
I am in favor of cleaning this up. The scattered initialization calls aren't nice to have. From what I understand, the public `import_array` macro was intended to be called directly from the extension initialization function. Burying the numpy initialization logic deeper in our code forced us to use the private function `_import_array`, instead.